src/Signature.ts

```diff
@@ -37,14 +43,52 @@ export class Signature {
         }
         const sigData = new Uint8Array([eosioRecoveryParam].concat(r, s));
         if (!ec) {
-            ec = constructElliptic(keyType);
+            if (keyType === KeyType.k1) {
+                ec = new EC('secp256k1');
+            } else {
+                ec = new EC('p256');
+            }
         }
         return new Signature({
             type: keyType,
             data: sigData,
         }, ec);
     }
 
+    /** Instantiate Signature from a Web Crypto Signature */
+    public static async fromWebCrypto(data: WebCryptoSignatureData, webCryptoSig: ArrayBuffer, publicKey: PublicKey, ec?: EC) {
+        if (!ec) {
+            ec = new EC('p256');
+        }
+
+        const hash = await crypto.subtle.digest('SHA-256', data);
+        const r = new BN(new Uint8Array(webCryptoSig.slice(0, 32)), 32);
+        let s = new BN(new Uint8Array(webCryptoSig.slice(32)), 32);
+        const halforder = ec.curve.n.ushrn(1); // shift right 1 bit -- division by two
+        if (s.ucmp(halforder) === 1) {
+            s = ec.curve.n.sub(s);
+        }
+        const recoveryParam = this.getRecoveryParam(Buffer.from(hash), {r, s}, publicKey.toString(), ec);
+        return Signature.fromElliptic({r, s, recoveryParam}, KeyType.r1, ec);
+    }
+
+    /** Replaced version of getRecoveryParam from `elliptic` library */
+    private static getRecoveryParam = (digest: BNInput, signature: EC.SignatureOptions, publicKey: string, ec: EC) => {
+        let recoveredKey: any;
+        for (let i = 0; i < 4; i++) {
+            try {
+                const keyPair = ec.recoverPubKey(digest, signature, i);
```

Comment:
Name `keyPair` is a bit confusing as private key is not here.

Reply:
Actually the `recoverPubKey` function will return an `ec` `keyPair` object, although the private will be empty and the public will be non-empty.